genesyscloud/tfexporter/genesyscloud_resource_exporter.go

```diff
@@ -1221,6 +1221,29 @@ func (g *GenesysCloudResourceExporter) sanitizeConfigMap(
 			} else {
 				configMap[key] = escapeString(val.(string))
 			}
+
+			// custom function to resolve the field to a specific data source depending on the value
+			if refAttrCustomResolver, ok := exporter.CustomAttributeResolver[currAttr]; ok {
```

Comment:
Oof.  This is a pretty deep and wide block of code.  Should we break this out into its own function and then have unit tests to test the logic?

Reply:
You're right. I didn't stop to consider it had gotten a bit out of control 😄 I'll refactor it